src/PressButtonLogic.js

```diff
@@ -0,0 +1,160 @@
+};
+const calculatePartialResult = (firstOperand, secondOperand, operator) => {
+    //console.log('secondOperand', secondOperand, operator);
+    switch (operator) {
+        case '+':
+            firstOperand += secondOperand;
+            return firstOperand;
+        case '-':
+            firstOperand -= secondOperand;
+            return firstOperand;
+        case 'x':
+            firstOperand *= secondOperand;
+            return firstOperand;
+        case '÷':
+            firstOperand /= secondOperand;
+            return firstOperand;
+        default:
+            return firstOperand;
+    }
+};
+
+const solveOrderOfPrecedence = state => {
+    //let newState = state;
+    if (
+        state.operators.array.length === state.result.length &&
+        state.operators.array.length > 1
+    ) {
+        state.operators.array.splice(0, 1);
+    }
+    for (let i = 0; i < state.operators.array.length - 1; i++) {
+        let val = state.operators.array[i];
+        if (val === 'x' || val === '÷') {
+            let result = state.result[i];
+            state.operators.array.splice(i, 1);
+            if (val === 'x') {
+                state.result[i + 1] *= result;
+            } else {
+                state.result[i + 1] = result / state.result[i + 1];
+            }
+
+            state.result.splice(i, 1);
+        }
+    }
+    return state;
+};
+
+const calculateResult = state => {
+    state = solveOrderOfPrecedence(state);
+    const lastElemResultArray = state.result.length - 1;
+    let lastElemOperatorsArray = state.operators.array.length - 1;
+    let finalResult = Number(state.result[0]);
+    if (lastElemResultArray + 1 > 1) {
+        if (state.operators.array.length === lastElemResultArray + 1) {
+            state.operators.array.splice(0, 1);
+            lastElemOperatorsArray = state.operators.array.length - 1;
+        }
+        for (let i = 0; i <= lastElemOperatorsArray; i++) {
+            finalResult = calculatePartialResult(
+                finalResult,
+                Number(state.result[i + 1]),
+                state.operators.array[i]
+            );
+        }
+        state.operators.array = [state.operators.array[lastElemOperatorsArray]];
+        state.operators.redundant = Number(state.result[lastElemResultArray]);
+    } else {
+        finalResult = Number(state.result[0]);
+        finalResult = calculatePartialResult(
+            finalResult,
+            state.operators.redundant,
+            state.operators.array[lastElemOperatorsArray]
+        );
+    }
+    state.default = true;
+    state.result = [finalResult.toString()];
+    return state;
+};
+
+const createNewDisplayingState = (newResult, newDefaultValue, state) => {
+    if (newResult === '.' && state.point) {
+        return state;
+    }
+    if (newResult === '.') state.point = true;
+    const saveResultArray = state.result;
+    const lastElem = saveResultArray.length - 1;
+    if (newDefaultValue === true) {
```

Comment:
Please try to add empty lines that group the code into logical blocks of code. It helps readability of the code.